modules/mysql/collect.go

```diff
@@ -64,6 +64,10 @@ func (m *MySQL) collect() (map[string]int64, error) {
 		}
 	}
 
+	if err := m.collectProcessListStatistics(collected); err != nil {
+		m.Errorf("error on collecting process list statistics: %v", err)
+	}
```

Comment:
I am not 100% sure, but [according to the requeste](https://github.com/netdata/netdata/issues/7879#issuecomment-581942235)r this query requires [PROCESS privileges](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_process).

I think we need to wrap it in another "if/else", like

https://github.com/netdata/go.d.plugin/blob/2181c601562b0d2fa9005756c3d141572f3b0758/modules/mysql/collect.go#L51-L57

But, as the TODO says, ideally we should stop doing the query only on "not enough permissions" error (if that is possible).

Reply:
I am doing a bit of research and found [this from MariaDB](https://mariadb.com/kb/en/show-processlist/#description):
>SHOW PROCESSLIST shows you which threads are running. You can also get this information from the information_schema.PROCESSLIST table or the mysqladmin processlist command. If you have the PROCESS privilege, you can see all threads. Otherwise, you can see only your own threads (that is, threads associated with the MariaDB account that you are using). If you do not use the FULL keyword, only the first 100 characters of each statement are shown in the Info field.

I tested on the latest MySQL and MariaDB from dockerhub and both give the same result.

- I created an anonymous user
- loged in from the terminal user `mysql -u anonym`
- executed `SELECT TIME,USER FROM INFORMATION_SCHEMA.PROCESSLIST 
WHERE Info IS NOT NULL AND Info NOT LIKE '%PROCESSLIST%' 
ORDER BY TIME`
- Got back an empty result set without errors. 
- executed `SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST`
```
*************************** 1. row ***************************
     ID: 11
   USER: anonym
   HOST: localhost
     DB: NULL
COMMAND: Query
   TIME: 0
  STATE: executing
   INFO: SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST
1 row in set (0.00 sec)
```


**Conclusion:**
It seems like the behaviour is confirmed and the user without the `process` privilege only gets back his own thread and in this was one executing `SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST`. 

If we want stats for all users, I think the only way to resolve this is to have another query at the beginning listing the privileges and checking if process is one of them for the current/netdata user**. Something like:
```
SELECT EXISTS(SELECT PRIVILEGE_TYPE FROM information_schema.USER_PRIVILEGES WHERE GRANTEE='''netdata''@''%''' and PRIVILEGE_TYPE='PROCESS') as HAS_PROCESS_PRIVILEGE;
```
Otherwise we will get the best, for all users if our user has the privilege or just metrics for our user.

Reply:
> If we want stats for all users, I think the only way to resolve this is to have another query at the beginning listing the privileges and checking if the process is one of them

What privileges do we need to check it? 😄 

---

Actually, that is a good idea. If the user has no needed privileges, there is no sense in doing that query. We need to log an info message and disable the query (not in this PR, I need to think about it).